### ISSUE TITLE
725: custom dependency groups?

### ISSUE THREAD

**Issue by tobiasdiez** [rgommers/peps#17]
PEP 735 (https://peps.python.org/pep-0735/), currently in draft, allows to specify dependency groups, e.g.
```
[dependency-groups]
test = ["pytest>7", "coverage"]
docs = ['sphinx']
```

Are there any plans to integrate with this, or provide a similar mechanism for external dependencies?
For example, `mathjax` dependency (via conda) for `docs`, some codecoverage cli tool for `tests` or github cli for `dev`.

**Comment by jaimergp** [comment]
I don't think we need to "integrate" with the (now approved) `[dependency-groups]` table, because it deals with PyPI dependencies. The current PEP 725 proposal has `external.optional-dependencies`, which might seem more akin to the `extras` feature`, but since none of the `[external]` table makes it to the distribute metadata, it's essentially a `[dependency-groups]` table for external metadata.

**Comment by tobiasdiez** [comment]
Sure, no deep integration is necessary.

What about using `external.dependency-groups` in place of `external.optional-dependencies` for symmetry?

**Comment by rgommers** [comment]
I think it makes sense to add `external.dependency-groups`, mirroring the structure and purpose of how it's used for PyPI dependencies. I think at this point that's very easy to do, and I can't think of a downside right now.

> since none of the `[external]` table makes it to the distribute metadata

This is one of the points I'm still least certain about, I think it could change during the discussion. Or we could still decide now to include it in one of two ways:
- add it to Core Metadata
- add it as a separate file (e.g., `external.json` - a lot simpler to deal with than Core Metadata)

**Comment by jaimergp** [comment]
I opened https://github.com/rgommers/peps/pull/30 with a potential solution to this issue.
